modules/data.atmosphere/R/ERA5_met_process.R

```diff
@@ -0,0 +1,188 @@
+  #getting site info
+  #getting site ID
+  observations <- c()
+  for (i in seq_along(1:length(settings$run))) {
+    command <- paste0("settings$run$settings.",i,"$site$id")
+    obs <- eval(parse(text=command))
+    observations <- c(observations,obs)
+  }
+  
+  #query site info
+  dbparms = list()
+  dbparms$dbname = "bety"
+  dbparms$host = "128.197.168.114"
+  dbparms$user = "bety"
+  dbparms$password = "bety"
+  
+  #Connection code copied and pasted from met.process
+  bety <- dplyr::src_postgres(dbname   = dbparms$dbname, 
+                              host     = dbparms$host, 
+                              user     = dbparms$user, 
+                              password = dbparms$password)
+  con <- bety$con
+  site_ID <- observations
+  suppressWarnings(site_qry <- glue::glue_sql("SELECT *, ST_X(ST_CENTROID(geometry)) AS lon,
+                                              ST_Y(ST_CENTROID(geometry)) AS lat FROM sites WHERE id IN ({ids*})",
+                                              ids = site_ID, .con = con))
+  suppressWarnings(qry_results <- DBI::dbSendQuery(con,site_qry))
+  suppressWarnings(qry_results <- DBI::dbFetch(qry_results))
+  site_info <- list(site_id=qry_results$id, site_name=qry_results$sitename, lat=qry_results$lat,
+                    lon=qry_results$lon, time_zone=qry_results$time_zone)
+  
+  #db query
+  if(Write){
+    mimetype <- "application/x-netcdf"
+    formatname <- "CF Meteorology"
+    hostname <- PEcAn.remote::fqdn()
+    # find mimetype, if it does not exist, it will create one
+    mimetypeid <- get.id("mimetypes", "type_string", mimetype, con, create = TRUE)
+    
+    # find appropriate format, create if it does not exist
+    formatid <- get.id(
+      table = "formats",
+      colnames = c("mimetype_id", "name"),
+      values = c(mimetypeid, formatname),
+      con = con,
+      create = TRUE,
+      dates = TRUE
+    )
+    
+    # setup parent part of query if specified
+    parent <- ""
+  }
+  
+  #loop over each site
+  #Record all IDs for inputs and dbfiles
+  if(Write){
+    Input_IDs <- list()
+  }
+  
+  #record paths of extracted .clim files if not writing into bety
+  if(!Write){
+    Clim_paths <- list()
+  }
+  
+  #loop over each site
+  for (i in 1:length(site_info$site_id)) {
+    #check if sub-folder exists, if doesn't then create a new folder specific for each site
+    site_outFolder <- paste0(out.path,'/',as.character(site_info$site_id[i]))
+    
+    #check if folder already exists, if it does, then jump to the next loop
+    if(!file.exists(site_outFolder)){
+      dir.create(site_outFolder)
+    }else{
+      print(paste0("The output files for site ",as.character(site_info$site_id[i])," already exists jump to the next site"))
+      next
+    }
+    
+    #Write into inputs table for each site
+    if(Write){
+      #insert into inputs table
+      cmd <- paste0(
+        "INSERT INTO inputs ",
```

Comment:
Do the Input Insert AFTER the met has been downloaded, not before, in case the download fails